main.c

```diff
@@ -119,12 +121,20 @@ static void _on_vmnet_packets_available(interface_ref iface, int64_t buf_count,
       received_count, buf_count);
   for (int i = 0; i < received_count; i++) {
     DEBUGF("[Handler i=%d] Sending to VDE: %ld bytes", i, pdv[i].vm_pkt_size);
-    ssize_t written =
+    while (1) {
+      ssize_t written =
         vde_send(vdeconn, pdv[i].vm_pkt_iov->iov_base, pdv[i].vm_pkt_size, 0);
-    DEBUGF("[Handler i=%d] Sent to VDE: %ld bytes", i, written);
-    if (written != (ssize_t)pdv[i].vm_pkt_size) {
-      perror("vde_send");
-      goto done;
+      if (written < 0 && errno == ENOBUFS) {
```

Comment:
Not directly related to the issue, but probably we can retry on EINTR too

Reply:
Did you ever see EINTR in the log files?  If this is rare enough, then we don't really need to worry, as the higher level transport will just retransmit the missing packet. The problem with ENOBUFS is that it happens **very** frequently.